### PR TITLE
python3Packages.simplicial: init at 1.0.1

### PR DESCRIPTION
**Pull request by zmberber**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by pinpox — APPROVED** [review]

File: pkgs/development/python-modules/simplicial/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    owner = "simoninireland";
+    repo = "simplicial";
+    rev = "v${version}";
+    hash = "sha256-o9YoswTthqW+xMrIrSMscRPnqz9A0EGQMXlbuIZ0OWE=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
+    pytestCheckHook
```
Comment (mweinelt) [inline pkgs/development/python-modules/simplicial/default.nix:26]: The `wheel` package is unnecessary here, as it is provided by the build-backend.

The `pytestCheckHook` needs to go into `nativeCheckinputs`. Please create that list below `pythonImportsCheck`.

File: pkgs/development/python-modules/simplicial/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  propagatedBuildInputs = [
+    matplotlib
+    numpy
+  ];
+
+  pythonImportsCheck = [ "simplicial" ];
+
+  meta = with lib; {
+    description = "Simplicial topology in Python";
+    homepage = "git@github.com:simoninireland/simplicial.git";
```
Comment (mweinelt) [inline pkgs/development/python-modules/simplicial/default.nix:38]:
```suggestion
    homepage = "https://github.com/simoninireland/simplicial";
```

Homepage links should be something people can navigate to.

File: pkgs/development/python-modules/simplicial/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  ];
+
+  propagatedBuildInputs = [
+    matplotlib
+    numpy
+  ];
+
+  pythonImportsCheck = [ "simplicial" ];
+
+  meta = with lib; {
```
Comment (mweinelt) [inline pkgs/development/python-modules/simplicial/default.nix:36]:
```suggestion
  meta = with lib; {
    changelog = "https://github.com/simoninireland/simplicial/blob/v${version}/HISTORY";
```

Setting up a pattern for the changelog URL allows tools to automatically pick that up on updates.

File: pkgs/development/python-modules/simplicial/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    description = "Simplicial topology in Python";
+    homepage = "git@github.com:simoninireland/simplicial.git";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/simplicial/default.nix:40]: As discussed we require maintainers for newly submitted packages. Please create your maintainer handle in a separate commit before this one, but still in this PR.

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Pull request closed without merging**